code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,223 @@
+/* 	< OH MY GOD. Can't you just make "/image/proc/foo()" instead of making these? >
+ * 		/appearance is a hardcoded byond type, and it is very internal type.
+ *		Its type is actually /image, but it isn't truly /image. We defined it as "/appearance"
+ * 		new procs to /image will only work to actual /image references, but...
+ * 		/appearance references are not capable of executing procs, because these are not real /image
```

Comment:
Appearances are byond's way of representing things that a client should draw. Icons and mutable appearances are things that can be drawn, which get converted into appearances internally so that they can be rendered. Images are icons/mutable appearances which are attached to a location, essentially a physical instance of an abstract mutable appearance.

Afaik, appearance is not a subtype of image.
https://www.byond.com/docs/ref/#/atom/var/appearance
https://www.byond.com/docs/ref/#/image/var

There are certain things which appearances have that images don't have:
- vis_flags
- text (used for text rendering mode)
- suffix
- opacity
- name
- mouse_opacity_var
- mouse_drop_zone_var
- mouse_over_pointer
- mouse_drag_pointer
- mouse_drop_pointer
- luminosity
- layer
- infra_luminosity
- invisibility
- gender
- desc

Since they don't appear to inherit from each other, I would manually enter the appearance vars instead of trying to infer them from image.

It seems like mutable appearances are closer to appearances, although they still differ.

Reply:
I guess /image does actually have these despite it being undocumented

Reply:
okay this is fine actually and seems to be a standard way of doing this